docs/Daedalus-Installer-Update-Procedure.md

```diff
@@ -0,0 +1,267 @@
+
+These values will also be summarised in the file `wiki.md` within the
+work dir.
+
+**Important**: If an update proposal is made with the wrong
+`applicationVersion`, the update mechanism will be useless until each
+user manually intervenes by manually installing an update.
+
+
+## 3. (Optional) Sign installer files with GPG
+
+This step requires a signing key available on the deployer host.
+
+    io -c NETWORK.yaml update-proposal sign-installers [-u signing.authority@iohk.io] DATE
+
+Without any key ID specified (`-u` option), it will sign with the
+default key in the default GPG keyring. If the signing key is
+protected with a passphrase, you will be prompted to enter it.
+
+This will place detached signatures in `.asc` files within the work
+dir. These will be uploaded to S3 at the same time as the installer
+files.
+
+
+## 4. Upload installer files to the S3 update bucket
+
+    io -c NETWORK.yaml update-proposal upload-s3 DATE
+
+This will hash the installer files with both `cardano-auxx` and
+`sha256sum`, then upload the hashed installers to S3.
+
+After this has run, inspect the `installerHashes` values in
+`params.yaml`.
+
+There will also be a file `daedalus-latest-version.json` added to the
+work dir with download links and SHA-256 hashes.
+
+*Hint*: The destination S3 bucket is configured with the
+`installer-bucket` value in `NETWORK.yaml`. You can edit this for
+debugging purposes.
+
+### Troubleshooting
+
+If you get an error like this: `AesonException "Error in $['mainnet_staging_short_epoch_wallet_win64']: key \"infra\" not present")`,
+ensure that the `cardano-sl-auxx` that you are using matches that
+cardano-sl version used by Daedalus.
+
+
+## 5. Update version JSON
+
+    io -c NETWORK.yaml update-proposal set-version-json DATE
+
+This will drop the previously created `daedalus-latest-version.json`
+into the S3 bucket. If done with the mainnet settings, it will have
+the effect of immediately updating the download links on
+[The Daedalus Wallet download page](https://daedaluswallet.io/#download).
+
+## 6. Propose the update and vote in favor using majority stake
+
+Find the IP address of a *privileged relay* with `io info`. This is
+normally private info so don't leak it.
+
+    io -c NETWORK.yaml update-proposal submit DATE --relay-ip 1.2.3.4 [--with-linux]
+
+By default, installers will be proposed for Windows and macOS, but not
+Linux. Use the `--with-linux` flag to include these installers in the
+update proposal.
+
+This will generate a new node db, copy keys from the top-level `keys`
+directory, then "rearrange" the copied keys.
+
+It will then send a transaction to the given relay.
+
+Note the proposal ID which is printed at the end of the output.
+
+If the update proposal was successful, the ratified proposal will take effect in 12 hours. See [Alex Vieth's](https://input-output-rnd.slack.com/archives/C2VJ41WDP/p1522951431000512) analysis of the code to corroborate the expected 12 hour delay.
```

Comment:
Is the slack.com link stable or should we inline the commen into a markdown file alongside thie repository as well?